src/main/scala/bitboard/OpaqueBitboard.scala

```diff
@@ -52,6 +52,20 @@ trait OpaqueBitboard[A](using A =:= Long) extends TotalWrapper[A, Long]:
     // remove the first non empty position
     def removeFirst: A = (a.value & (a.value - 1L)).bb
 
+    // check if the intersection of all bitboards is not empty
+    def sharedAny(o: Long, xs: Long*): Boolean =
+      (a & xs.foldLeft(o)(_ & _)).nonEmpty
+
+    // check if the intersection of all bitboards is not empty
+    def sharedAny[B](o: B, xs: B*)(using sr: BitboardRuntime[B]): Boolean =
+      (a & xs.foldLeft(sr(o))(_ & sr(_))).nonEmpty
```

Comment:
why  `foldLeft` rather than `fold`?

Reply:
I remember that `foldLeft` is for linear calculation, and `fold` is for non-linear calculation. Usually I think we should use `foldLeft` as default.